controllers/snapshot/snapshot_adapter.go

```diff
@@ -396,6 +422,11 @@ func (a *Adapter) EnsureAllReleasesExist() (controller.OperationResult, error) {
 			a.snapshot, h.LogActionUpdate)
 		return controller.RequeueOnErrorOrStop(a.client.Status().Patch(a.context, a.snapshot, patch))
 	}
+	startTime, ok := GetAppStudioTestsFinishedTime(a.snapshot)
```

Comment:
This looks like the right place

Reply:
actually it's not, when we rerun reconciliation, this may generate wrong statistics, it should be recorded only when release is created, we cannot tell from this place

Reply:
but recorded only once, if at least one release has been created